File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.ts
function makeEqual(words: string[]): boolean {
    // Initialize frequency array for 26 lowercase English letters
    const freq = new Array(26).fill(0);

    // Count the frequency of each character in all words
    for (let word of words) {
        for (let char of word) {
            freq[char.charCodeAt(0) - 'a'.charCodeAt(0)]++;
        }
    }
    
    // Check if each character's count is divisible by the number of words
    for (let count of freq) {
        if (count % words.length !== 0) {
            return false;
        }
    }
    
    return true;
}

File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        freq = [0 for i in range(26)]

        for word in words:
            for char in word:
                freq[ord(char) - ord('a')] += 1
        
        for count in freq:
            if count % len(words) != 0:
                return False
        return True

        